File: Cpp/ZigZag Conversion.cpp
// 35ms, 27.56%
class Solution
{
public:
    string convert(string s, int numRows)
    {
        if(s.empty() || s.length() == 1 || numRows == 1)
            return s;
        string ans[numRows];
        for(auto x : ans) x = "";
        
        bool incre = true;
        for(int idx = 0, row = 0; idx < s.length(); ++idx)
        {
            ans[row] += s[idx];
            row += incre ? 1 : -1;
            if(row == numRows - 1) incre = false;
            else if(row == 0) incre = true; 
        }
        for(int row = 1; row < numRows; ++row)
            ans[0] += ans[row];
        
        return ans[0];
    }
};

File: Python3/ZigZag Conversion.py
# 100ms, 100.0%
class Solution:
    def convert(self, s, numRows):
        if numRows == 1: return s
        index, is_down, res = 0, 1, [[] for i in range(numRows)]
        for i in range(len(s)):
            if index == 0: is_down = 1
            elif index == numRows - 1: is_down = -1
            res[index].append(s[i])
            index += is_down
        return ''.join([''.join(part) for part in res])
    
